src/py42/clients/settings/device_settings.py

```diff
@@ -0,0 +1,415 @@
+                self._orig_destinations, self.destinations
+            )
+
+    def lock_destination(self, destination_guid):
+        """Locks an in-use destination, disallowing the device owner from removing this
+        destination from their backup. Raises an exception if the supplied destination
+        guid is not in use on this backup set, or not available to the parent device/org.
+        """
+        destination_guid = str(destination_guid)
+        if destination_guid in self._manager.available_destinations:
+            if destination_guid not in self.destinations:
+                raise Py42Error("Destination is not added to device, unable to lock.")
+            else:
+                for d in self.data["destinations"]:
+                    if d["@id"] == destination_guid:
+                        d["@locked"] = "true"
+                self._changes["destinations"] = show_change(
+                    self._orig_destinations, self.destinations
+                )
+        else:
+            raise Py42Error(
+                "Invalid destination guid or destination not offered to device's Org."
+            )
+
+    def unlock_destination(self, destination_guid):
+        """Unlocks an in-use destination, allowing the device owner to remove this
+        destination from their backup. Raises an exception if the supplied destination
+        guid is not in use on this backup set, or not available to the parent device/org.
+        """
+        destination_guid = str(destination_guid)
+        self._raise_if_invalid_destination(destination_guid)
+        if destination_guid not in self.destinations:
+            raise Py42Error("Destination is not added to device, unable to unlock.")
+        else:
+            for d in self.data["destinations"]:
+                if d["@id"] == destination_guid:
+                    del d["@locked"]
+            self._changes["destinations"] = show_change(
+                self._orig_destinations, self.destinations
+            )
+
+    def _raise_if_invalid_destination(self, destination_guid):
+        if destination_guid not in self._manager.available_destinations:
+            raise Py42Error(
+                "Invalid destination guid or destination not offered to device's Org."
+            )
+
+    def _extract_file_selection_lists(self):
+        try:
+            pathset = self.data["backupPaths"]["pathset"][0]["path"]
+        except KeyError:  # no files selected
+            return [], []
+        if isinstance(pathset, dict):
+            pathset = [pathset]
+        includes = [p["@include"] for p in pathset if "@include" in p]
+        excludes = [p["@exclude"] for p in pathset if "@exclude" in p]
+        return includes, excludes
+
+    def _build_file_selection(self):
+        if not self._included_files:
+            self.data["backupPaths"]["pathset"] = {
+                "paths": {"@cleared": "true", "@os": "Linux", "path": []}
```

Comment:
Why is the Linux the default?

Reply:
I'm not sure, but that's what the UI uses regardless of what OS the client actually is installed on. I'm guessing it was initially designed to allow for the server to handle the paths differently based on OS, but they eventually had the client do all conversions (i.e. `C:/Users` > `C:\Users` on Windows).